src/BrightScriptCompletionItemProvider.ts

```diff
@@ -8,23 +8,136 @@ import {
+        ifAudioGuide: ifAudioGuideCompletionItems,
+        ifAudioMetadata: ifAudioMetadataCompletionItems,
+        ifAudioPlayer: ifAudioPlayerCompletionItems,
+        ifAudioResource: ifAudioResourceCompletionItems,
+        ifByteArray: ifByteArrayCompletionItems,
+        ifChannelStore: ifChannelStoreCompletionItems,
+        ifCompositor: ifCompositorCompletionItems,
+        ifDateTime: ifDateTimeCompletionItems,
+        ifDeviceInfo: ifDeviceInfoCompletionItems,
+        ifDraw2D: ifDraw2DCompletionItems,
+        ifEnum: ifEnumCompletionItems,
+        ifEVPCipher: ifEVPCipherCompletionItems,
+        ifEVPDigest: ifEVPDigestCompletionItems,
+        ifFileSystem: ifFileSystemCompletionItems,
+        ifFont: ifFontCompletionItems,
+        ifFontRegistry: ifFontRegistryCompletionItems,
+        ifHdmiStatus: ifHdmiStatusCompletionItems,
+        ifHMAC: ifHMACCompletionItems,
+        ifHttpAgent: ifHttpAgentCompletionItems,
+        ifImageMetadata: ifImageMetadataCompletionItems,
+        ifList: ifListCompletionItems,
+        ifLocalization: ifLocalizationCompletionItems,
+        ifMessagePort: ifMessagePortCompletionItems,
+        ifMicrophone: ifMicrophoneCompletionItems,
+        ifPath: ifPathCompletionItems,
+        ifRegex: ifRegexCompletionItems,
+        ifRegion: ifRegionCompletionItems,
+        ifRegistry: ifRegistryCompletionItems,
+        ifRegistrySection: ifRegistrySectionCompletionItems,
+        ifRSA: ifRSACompletionItems,
+        ifScreen: ifScreenCompletionItems,
+        ifSocketAddress: ifSocketAddressCompletionItems,
+        ifSocketAsync: ifSocketAsyncCompletionItems,
+        ifSocketCastOption: ifSocketCastOptionCompletionItems,
+        ifSocket: ifSocketCompletionItems,
+        ifSocketConnection: ifSocketConnectionCompletionItems,
+        ifSocketConnectionOption: ifSocketConnectionOptionCompletionItems,
+        ifSocketConnectionStatus: ifSocketConnectionStatusCompletionItems,
+        ifSocketOption: ifSocketOptionCompletionItems,
+        ifSocketStatus: ifSocketStatusCompletionItems,
+        ifSourceIdentity: ifSourceIdentityCompletionItems,
+        ifSprite: ifSpriteCompletionItems,
+        ifStringOps: ifStringOpsCompletionItems,
+        ifSystemLog: ifSystemLogCompletionItems,
+        ifTextToSpeech: ifTextToSpeechCompletionItems,
+        ifTextureManager: ifTextureManagerCompletionItems,
+        ifTextureRequest: ifTextureRequestCompletionItems,
+        ifTimespan: ifTimespanCompletionItems,
+        ifUrlTransfer: ifUrlTransferCompletionItems,
+        ifVideoPlayer: ifVideoPlayerCompletionItems,
+        ifXMLElement: ifXMLElementCompletionItems,
+        ifXMLList: ifXMLListCompletionItems
+    };
 
     public provideCompletionItems(document: TextDocument, position: Position, token: CancellationToken, context: vscode.CompletionContext): CompletionItem[] {
-        //TODO - do something useful here!
-        return this.BuiltinCompletionItems;
```

Comment:
I'm deeply concerned by this - this looks to me like it won't _augment_ the existing contextual completions; but will *remove them* - that would mean we'd lose all the completion hints for our own vars and functions, which would greatly degrade the experience - I spent most my time calling my own method's and using my vars, not the interface ones, so I see nothing but pain if I lose the builtin completion items, which is why I didn't mess with this when I did the other definition and symbol work.

Reply:
In the context of this PR, we do want to throw away completions in this case, because you have specifically typed the full name of the interface after your variable, (i.e. `yourVar.ifArray`) meaning there are no other valid completions except the ones that are provided in this PR.